zio-query/shared/src/main/scala/zio/query/ZQuery.scala

```diff
@@ -833,6 +865,27 @@ final class ZQuery[-R, +E, +A] private (private val step: ZIO[R, Nothing, Result
         case (_, Result.Fail(e))                                => Result.fail(e)
       }
     }
+
+  /**
+   * These methods can improve UX and performance in some cases, but when used
+   * improperly they can lead to unexpected behaviour in the application code.
+   *
+   * Make sure you really understand them before using them!
+   */
+  def unsafe: UnsafeApi = new UnsafeApi {}
```

Comment:
Instead of having an `unsafe` method on the query itself, should this perhaps be moved to the companion object, and have something like this:

```scala
val query = ???
val memoizedQuery = ZQuery.unsafe.memoize(query)
```